src/manager.js

```diff
@@ -70,6 +71,10 @@ class Manager extends EventEmitter {
   async watch (name, options, callback) {
     options.newJobCheckInterval = options.newJobCheckInterval || this.config.newJobCheckInterval
 
+    const teamQueue = options.batchSize ? null : new PQueue({ concurrency: options.teamConcurrency || 1, timeout: 15 * 60 * 1000 })
+    const teamSize = options.teamSize || 1
+    const queueSize = () => teamQueue.size - teamQueue.pending
```

Comment:
Thinking about this (for probably too long, too).  Following is an example I'm using to reason about the flow.

1. subscribe() with teamSize = 10 and concurrency = 2, teamMinimumFetch = undefined
2. First load starts 2 promises immediately
  * queue.size = 8
  * queue.pending = 2
  * queueSize() = 6
  * teamSize - queueSize() = 4
3. First promise is resolved, on('next') fires.
  * queue.size = 7
  * queue.pending = 2
  * queueSize() = 5
  * teamSize - queueSize() = 5
4. Fetch 5 and load into the queue
  * queue.size = 12
  * queue.pending = 2
  * queueSize() = 10
  * teamSize - queueSize() = 0

Based on this, I think this might load more jobs into the queue that we want.  What if we were to use a simple "if the queue is <= teamSize / 2 then fetch the difference" instead of a config setting?  

Also, can you tell me why you're subtracting the pending promises from the size and not adding them together to get the size?

Reply:
ah, this is a bad copy and paste from when it was inline in the previous version, it was: 
`numToFetch = options.teamSize - teamQueue.size - teamQueue.pending`

it should be teamQueue.size + teamQueue.pending

Fixing that should mean it'll maintain the queue at teamSize.

Would setting it to teamSize / 2 be desirable for all cases?
If there were folks that were running their teamConcurrency at greater than teamSize / 2, it would mean the queue would be spending some unintended idle time.

Reply:
Yeah, I agree that it shouldn't be allowed to stay under the teamConcurrency value.  When I first read the logic, I didn't realize the `off()` would relieve some pressure as these events could be ignored during the next job fetch and we'd catch up later if jobs were completing quickly.  I think it would probably be just fine the teamMinimumFetch config property for now.

Reply:
Ok, just want to check I'm following:
When you say leave out teamMinimumFetch and also remove the teamSize/2 we just have it return to the loop after at least 1 job returns?

Reply:
Yes

Reply:
Ok, i've removed teamMinimumFetch and it will always fetch if the queue drops below the team size